Directory.Build.props

```diff
@@ -7,7 +7,7 @@
   </PropertyGroup>
 
   <PropertyGroup>
-    <LangVersion>8</LangVersion>
+    <LangVersion>12</LangVersion>
```

Comment:
This concerns me related to targeting net462, since it might introduce language usage not supported by that framework. What are your thoughts on it?

Reply:
There is a side effect in .NET 8 due to this commit: dotnet/runtime#89736. We can no longer pass a ref parameter to a method expecting an in parameter. Otherwise the following error occurs:

`src\MassTransit.Abstractions\NewId\NewIdFormatters\IntrinsicsHelper.cs(18,50): error CS9194: Argument 2 may not be passed with the 'ref' keyword in language version 8.0. To pass 'ref' arguments to 'in' parameters, upgrade to language version 12.0 or greater.`

For me, increasing the version of the language used should have no impact. As long as the project compiles when targeting net462, I think we can consider that it's ok. I'm open to other opinions.